### PR TITLE
add tls functionality

### PR DESCRIPTION
**Pull request by lucian-vanghele**
it is using WEB toolkit from Prometheus community in order to allow https option for the /metrics endpoint.

### CODE REVIEW

**Comment by burningalchemist** [comment]
Hey @lucian-vanghele, thanks for the PR, the idea is brilliant. :+1:

I think we might need to apply several changes, though. Let's address branch conflicts first: packages `prometheus/common` and `snowflakedb/gosnowflake` are slightly out of date in your end.

File: cmd/sql_exporter/main.go [hunk 1]
```diff
@@ -77,8 +80,14 @@ func main() {
 	if *enableReload {
 		http.HandleFunc("/reload", reloadCollectors(exporter))
 	}
-	klog.Infof("Listening on %s", *listenAddress)
-	klog.Fatal(http.ListenAndServe(*listenAddress, nil))
+
+	promlogConfig := &promlog.Config{}
+	logger := promlog.New(promlogConfig)
```
Comment (burningalchemist) [inline cmd/sql_exporter/main.go:85]: I see that `exporter-toolkit/web` is using `go-kit/log` for logging, while this project currently is based upon `klog/v2`. I think we might eventually want to try to migrate it to `go-kit/log`, but it's out of scope for this pull request.

I believe we can follow the path of [Prometheus](https://github.com/LeviHarrison/prometheus/commit/9bb3555fe4cf2f1d93d2f1d665163ec18365af21) and use [klog-gokit](https://github.com/simonpasquier/klog-gokit#usage) package as a bridge between these two.

It's a minor change, so shouldn't take long. Please let me know if you need any assistance, I'm happy to help. 👍

**Review by burningalchemist — CHANGES_REQUESTED** [review]
Some small changes are requested, but overall looking forward to merging it. 😉

File: cmd/sql_exporter/main.go [hunk 2]
```diff
@@ -77,8 +80,14 @@ func main() {
...
+	klog.Info("Listening on", *listenAddress)
+	server := &http.Server{Addr: *listenAddress}
+	if err := web.Listen(server, *webcfgFile, logger); err != nil {
```
[the remark this answers is not in the document]
Comment (lucian-vanghele) [reply cmd/sql_exporter/main.go:88]: done

File: cmd/sql_exporter/main.go (hunk 1 above)
Comment (lucian-vanghele) [reply cmd/sql_exporter/main.go:85]: I prefered to got for go-kit/log; I had some issues using klog-gokit

Comment (lucian-vanghele) [reply cmd/sql_exporter/main.go]: it complains about the the argument's type of klog.SetLogger method
honestly, I don't know what to do now ... so better if you have some time and check

Comment (burningalchemist) [reply cmd/sql_exporter/main.go]: @lucian-vanghele, Sure! I'll take a look and keep you posted. 👍😃

Comment (burningalchemist) [reply cmd/sql_exporter/main.go]: @lucian-vanghele I've taken a look, it's up and running. I'll try to push changes to your PR in the coming days.

**Comment by burningalchemist** [comment]
@lucian-vanghele could you allow me to make changes on your upstream branch [(link)](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork), so we could merge this particular pull request? Otherwise I can open a separate one. Let me know. :)

**Comment by burningalchemist** [comment]
This is closed in favour of #124.
@lucian-vanghele Thank you for the idea and your contribution! :+1:

**Pull request closed without merging**